zc_common/remote_resource/tests.py

```diff
@@ -12,11 +12,54 @@ class ResponseTestCase(APITestCase):
     FAILURE_HIGH_LEVEL_KEYS = ['errors']
     FAILURE_DATA_KEYS = ['status', 'source', 'detail']
 
+    INSTANCE_TOP_LEVEL_KEYS = ["type", "id", "attributes"]
+
     def convert_to_list(self, data):
-        if isinstance(data, list):
+        if isinstance(data, dict):
+            return [data]
+        elif isinstance(data, list):
             return data
         else:
-            return [data]
+            return data.all()
+
+    def validate_instance_in_data(self, data, instance, attributes, relationship_keys=None):
+        self.assertTrue(all(key in data for key in self.INSTANCE_TOP_LEVEL_KEYS))
+
+        instance_attributes = data["attributes"]
+        for key in attributes:
+            self.assertEqual(instance_attributes[camelize(key, False)], getattr(instance, key))
+
+        if not relationship_keys:
+            return True
```

Comment:
The boolean here is just a dummy value, safer to return True than False. However, early return is a good coding practice. Indenting under an if reduces codes readability.